biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -353,413 +378,531 @@ subroutine spawn_patches( currentSite, bc_in)
 
        ! Only create new patches that have non-negligible amount of land
        if((currentPatch%area*currentPatch%disturbance_rate) > nearzero ) then
-          site_areadis = site_areadis + currentPatch%area * currentPatch%disturbance_rate
+          
+          ! figure out whether the receiver patch for disturbance from this patch will be 
+          ! primary or secondary land receiver patch is primary forest only if both the 
+          ! donor patch is primary forest and the dominant disturbance type is not logging
+          if ( currentPatch%anthro_disturbance_label .eq. primaryforest .and. &
+               (currentPatch%disturbance_rates(dtype_ilog) .lt. currentPatch%disturbance_rates(dtype_ifall) .or. &
+               currentPatch%disturbance_rates(dtype_ilog) .lt. currentPatch%disturbance_rates(dtype_ifire)) ) then
+             
+             site_areadis_primary = site_areadis_primary + currentPatch%area * currentPatch%disturbance_rate
+          else
+             site_areadis_secondary = site_areadis_secondary + currentPatch%area * currentPatch%disturbance_rate          
+          endif
+          
        end if
-
+       
        currentPatch => currentPatch%older     
-
+       
     enddo ! end loop over patches. sum area disturbed for all patches. 
-
-    if (site_areadis > nearzero) then  
-
+    
+    ! It is possible that no disturbance area was generated
+    if ( (site_areadis_primary + site_areadis_secondary) > nearzero) then  
+       
        cwd_ag_local = 0.0_r8
        cwd_bg_local = 0.0_r8
        leaf_litter_local = 0.0_r8
        root_litter_local = 0.0_r8
        age = 0.0_r8
 
-       allocate(new_patch)
-       call create_patch(currentSite, new_patch, age, site_areadis, &
-            cwd_ag_local, cwd_bg_local, leaf_litter_local, &
-            root_litter_local, bc_in%nlevsoil)
-
-       new_patch%tallest  => null()
-       new_patch%shortest => null()
+       ! create two empty patches, to absorb newly disturbed primary and secondary forest area
+       ! first create patch to receive primary forest area
+       if ( site_areadis_primary .gt. nearzero ) then
+          allocate(new_patch_primary)
+          call create_patch(currentSite, new_patch_primary, age, site_areadis_primary, &
+               cwd_ag_local, cwd_bg_local, leaf_litter_local, &
+               root_litter_local, bc_in%nlevsoil, primaryforest)
+          new_patch_primary%tallest  => null()
+          new_patch_primary%shortest => null()
+       endif
 
+       ! next create patch to receive secondary forest area
```

Comment:
Checked through to see that age is set to zero for the new secondary patch, and that age sets the age since anthro disturbance, and that angtho age is updated when secondary patches donate area through natural means.  If they donate through a logging event, then they contribute a zero to the weighted age of the new secondary patch.   Makes sense to me.